zschema/keys.py

```diff
@@ -21,34 +26,42 @@ def __init__(self, port):
         self.port = str(port)
 
     def to_bigquery(self):
-        return "p%s" % self.port
+        return "p{:s}".format(self.port)
 
     def to_es(self):
         return self.port
 
     def to_proto(self):
-        return "p%s" % self.port
+        return "p{:s}".format(self.port)
 
     to_string = to_es
 
     def __eq__(self, other):
-        if type(other) == int:
-            return int(self.port).__eq__(other)
-        elif type(other) in (str, unicode):
-            return self.port.__eq__(str(other))
+        if isinstance(other, int):
+            return int(self.port) == other
+        elif isinstance(other, string_types):
+            return self.port == other
         else:
-            return self.port.__eq__(other.port)
+            return self.port == other.port
 
     def __hash__(self):
         return self.port.__hash__()
 
-    def __cmp__(self, other):
-        if type(other) == int:
-            return cmp(int(self.port), other)
-        elif type(other) in (str, unicode):
-            return cmp(self.port, str(other))
+    def __lt__(self, other):
+        if isinstance(other, int):
+            return self.port < str(other)
+        elif isinstance(other, string_types):
```

Comment:
I realize this keeps the old behavior...but do we really want to sort port numbers as strings?

Reply:
Probably not, but I didn't want to make that change here. Also, if we compare them as integers it gets complicated if we're comparing them against a non-integer string, such as `"example"`